### PR TITLE
[CORE-1646] add config management to CLI;

### PR DESCRIPTION
**Pull request by opalczynski**

### CODE REVIEW

**Comment by opalczynski** [comment]
@Zhebr ? :) Can you?

File: syncano_cli/config_commands/command.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        click.echo('Variable `{}` set to `{}`.'.format(name, value))
+
+    def delete(self, name):
+        config = self.instance.get_config()
+        if name in config:
+            config.pop(name)
+        else:
+            click.echo('ERROR: variable `{}` not found.'.format(name))
+            sys.exit(1)
+        self.instance.set_config(config)
```
Comment (kotchaosu) [inline syncano_cli/config_commands/command.py:42]: there is a pattern

```
config = self.instance.get_config()
# something
self.instance.set_config(config)
click.echo(message)
```

decorator?

File: syncano_cli/config_commands/commands.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    try:
+        connection = create_connection(config)
+        instance = connection.Instance.please.get(name=instance_name)
+        config_command = ConfigCommand(instance=instance)
+        ctx.obj['config_command'] = config_command
+        if ctx.invoked_subcommand is None:
+            config_command.config_show()
+
+    except Exception as e:
+        click.echo(u'ERROR: {}'.format(e))
```
Comment (kotchaosu) [inline syncano_cli/config_commands/commands.py:33]: why not raising specific exceptions?

File: syncano_cli/config_commands/commands.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+@click.pass_context
+@click.argument('name')
+def delete(ctx, name):
+
+    config_command = ctx.obj['config_command']
+
+    try:
+        config_command.delete(name)
+
+    except Exception as e:
```
Comment (kotchaosu) [inline syncano_cli/config_commands/commands.py:79]: decorator?

File: tests/test_config_commands.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        result = self._add_config('api-key', '981xx')
+        self.assertIn('Variable `api-key` set to `981xx`.', result.output)
+
+        result = self._list_config()
+        self.assertIn('api-key', result.output)
+        self.assertIn('981xx', result.output)
+
+        # test if ERROR is displayed;
+        result = self._add_config('api-key', '981xx')
+        self.assertIn('already in config', result.output)
```
Comment (kotchaosu) [inline tests/test_config_commands.py:26]: that's why I asked about exceptions - now you need change error message in two palces

File: tests/test_config_commands.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+        # test multiple add;
+        self._add_config('api-key-ui', '981xxy')
+        result = self._list_config()
+        self.assertIn('api-key', result.output)
+        self.assertIn('981xx', result.output)
+        self.assertIn('api-key-ui', result.output)
```
Comment (kotchaosu) [inline tests/test_config_commands.py]: why are you checking partial and full key-value names?

File: syncano_cli/config_commands/command.py (hunk 1 above)
Comment (opalczynski) [reply syncano_cli/config_commands/command.py:42]: I know that CLI have such issues - well it's evolving :) and have in mind that such things will be improved in the future.

File: syncano_cli/config_commands/commands.py (hunk 2 above)
Comment (opalczynski) [reply syncano_cli/config_commands/commands.py:33]: Basically - I think that showing a stacktrace to the end user is not rly nice. This is why the Exception is catched - which I know is not considered as a 'good practice'.

File: syncano_cli/config_commands/commands.py (hunk 3 above)
Comment (opalczynski) [reply syncano_cli/config_commands/commands.py:79]: As above - the whole structure of the CLI 'in the way' appears in the last week, after some rework, need re-work it on the all of the CLI commands - will be done in separate task.

File: tests/test_config_commands.py (hunk 5 above)
Comment (opalczynski) [reply tests/test_config_commands.py]: Yep - should change this test, basically - it should check if new and old values are present in the output. Thanks. Nice catch.

File: syncano_cli/config_commands/command.py (hunk 1 above)
Comment (kotchaosu) [reply syncano_cli/config_commands/command.py:42]: ok then

File: tests/test_config_commands.py (hunk 4 above)
Comment (opalczynski) [reply tests/test_config_commands.py:26]: Yep, should define a better ERROR handling in CLI :) Aware of that.

File: syncano_cli/config_commands/commands.py (hunk 2 above)
Comment (kotchaosu) [reply syncano_cli/config_commands/commands.py:33]: I'm just saying about two things:
- knowing what we're catching
- eventual changing in error messages names

File: syncano_cli/config_commands/commands.py (hunk 3 above)
Comment (kotchaosu) [reply syncano_cli/config_commands/commands.py:79]: ok

File: tests/test_config_commands.py (hunk 4 above)
Comment (kotchaosu) [reply tests/test_config_commands.py:26]: Ok, I understand it's cross-project task. If you plan to do so - great. Just wanted to point that out.

**Pull request merged**